From 04439790ce953a9ff6112350267980578fc5e490 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:18 -0400
Subject: [tools] perf/tests/x86: Fix memory leak in sample_ustack()

Message-id: <1410181752-25631-134-git-send-email-jolsa@redhat.com>
Patchwork-id: 91791
O-Subject: [PATCH RHEL7.1 BZ1134356 133/307] perf tests x86: Fix memory leak in sample_ustack()
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 763d7f5f2718f085bab5a9e63308349728f3ad12
Author: Masanari Iida <standby24x7@gmail.com>
Date: Sun Apr 20 00:16:41 2014 +0900

description
===========
The buf is not freed, when kernel failed to get stack map
and return.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/arch/x86/tests/dwarf-unwind.c b/tools/perf/arch/x86/tests/dwarf-unwind.c
index b602ad9..b8c0102 100644
--- a/tools/perf/arch/x86/tests/dwarf-unwind.c
+++ b/tools/perf/arch/x86/tests/dwarf-unwind.c
@@ -26,6 +26,7 @@ static int sample_ustack(struct perf_sample *sample,
  map = map_groups__find(&thread->mg, MAP__FUNCTION, (u64) sp);
  if (!map) {
   pr_debug("failed to get stack map\n");
+  free(buf);
   return -1;
  }
 
-- 
1.7.1